app/test/integration_test.go

```diff
@@ -344,3 +346,24 @@ func (s *IntegrationTestSuite) TestShareInclusionProof() {
 		require.NoError(t, blobProof.Validate(blockRes.Block.DataHash))
 	}
 }
+
+// ExtendBlobTest re-extends the block and compares the data roots to ensure
+// that the public functions for extending the block are working correctly.
+func (s *IntegrationTestSuite) ExtendBlobTest(t *testing.T, block *coretypes.Block) {
+	eds, err := app.ExtendBlock(block.Data, block.Header.Version.App)
+	require.NoError(t, err)
+	dah := da.NewDataAvailabilityHeader(eds)
+	require.Equal(t, dah.Hash(), block.DataHash.Bytes())
+}
+
+func (s *IntegrationTestSuite) TestEmptyBlock() {
+	t := s.T()
+	emptyHeights := []int64{1, 2, 3}
+	for _, h := range emptyHeights {
+		blockRes, err := s.cctx.Client.Block(s.cctx.GoContext(), &h)
+		require.NoError(t, err)
+		require.True(t, app.EmptyBlock(blockRes.Block.Data, blockRes.Block.Header.Version.App))
+		s.ExtendBlobTest(t, blockRes.Block)
+	}
+
+}
```

Comment:
testing these via integration tests is likely a bit overkill, however since they are public functions this should give us a high degree of confidence that they will work as intended